pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -573,4 +580,62 @@ public int getHllLog2m() {
     return _hllLog2m;
   }
 
+  /**
+   * Converts segment metadata to json
+   * @param columnFilter list only  the columns in the set. Lists all the columns if
+   *                     the parameter value is null
+   * @return json representation of segment metadata
+   */
+  public JSONObject toJson(Set<String> columnFilter) {
+
```

Comment:
You are right. That's the route I tried first but this class is not constructed for that. We end up polluting with annotations and custom setters (and getters may not work).